### PR TITLE
PitchShiftEffect: add description comments

### PR DESCRIPTION
**Pull request by davidchocholaty**
This PR adds the comments, primarily for the usage
of the RubberBand API, for the Pitch shift effect processing.

### CODE REVIEW

File: src/effects/backends/builtin/pitchshifteffect.cpp [hunk 1]
```diff
@@ -134,19 +136,30 @@ void PitchShiftEffect::processChannel(
             m_currentFormant != formantPreserving) {
         m_currentFormant = formantPreserving;
 
+        // Set the RubberBand option for the formant preserving.
```
Comment (daschuer) [inline src/effects/backends/builtin/pitchshifteffect.cpp:139]: This comment just repeats the code.

File: src/effects/backends/builtin/pitchshifteffect.cpp [hunk 2]
```diff
@@ -134,19 +136,30 @@ void PitchShiftEffect::processChannel(
...
         pState->m_pRubberBand->setFormantOption(m_currentFormant
                         ? RubberBand::RubberBandStretcher::
                                   OptionFormantPreserved
                         : RubberBand::RubberBandStretcher::
                                   OptionFormantShifted);
     }
 
+    // Get the parameter value of the Pitch knob
+    // based on the current setting of the Range knob.
```
Comment (daschuer) [inline src/effects/backends/builtin/pitchshifteffect.cpp:148]: This comment does also not add any value. 
You may explain that one of from 0...X and the other from -x to +y ... and describe the implications.

File: src/effects/backends/builtin/pitchshifteffect.cpp [hunk 3]
```diff
@@ -134,19 +136,30 @@ void PitchShiftEffect::processChannel(
...
+    // 1. Semitones mode (true) - The Pitch knob parameter value is recalculated
+    //    to proceed on the semitone chromatic scale. By default,
+    //    this mode is used.
+    // 2. Continuous mode (false) - The Pitch knob changes values continuously,
+    //    the same as the RubberBand classic approach.
     if (m_pSemitonesModeParameter->toBool()) {
         pitchParameter = roundToFraction(pitchParameter, kSemitonesPerOctave);
     }
 
+    // Calculate the result pitch setting for the RubberBand.
```
Comment (daschuer) [inline src/effects/backends/builtin/pitchshifteffect.cpp:162]: Here also no value. You may tell us why std::pow(2.0 is required.

**Review by daschuer — CHANGES_REQUESTED** [review]
I have left some comments.  Sorry if they look rude, that was not meant like that. 
I am happy about such Document PRs, Thank you.  

It is just, that I share the opinion that comments are only a last resort to explain difficult code parts in cases it was not possible to put the information into the code itself.

That helps that the reader does not waste time reading it and limit the risk that the code comment is rotting after refactoring.  

In this case, think about the pitfalls you have managed and write about them.

**Comment by davidchocholaty** [comment]
Absolutely no problem, thank you very much for the advice and proposals. I will do it this way.

**Review by daschuer — APPROVED** [review]
Thank you for all the additional details :-)

**Comment by davidchocholaty** [comment]
Thank you for merging. :+1:

**Pull request merged**